api_app/analyzers_manager/observable_analyzers/ailtyposquatting.py

```diff
@@ -0,0 +1,41 @@
+import logging
+import math
+
+from ail_typo_squatting import runAll
+from ail_typo_squatting.dns_local.resolving import dnsResolving
+
+from api_app.analyzers_manager import classes
+
+logger = logging.getLogger(__name__)
+
+
+class AilTypoSquatting(classes.ObservableAnalyzer):
+    """
+    wrapper for https://github.com/typosquatter/ail-typo-squatting
+    """
+
+    dns_resolving: bool = False
+
+    def update(self) -> bool:
+        pass
+
+    def run(self):
+        response = {}
+        logger.info(
+            f"""running AilTypoSquatting on {self.observable_name}
+                    with tlp {self._job.tlp}
+                    and dns resolving {self.dns_resolving}"""
+        )
+        resultList = []
+        response["algorithms"] = runAll(
+            domain=self.observable_name,
+            limit=math.inf,
+            formatoutput="yara",
+            pathOutput=None,
+        )
+        if self._job.tlp == "CLEAR" and self.dns_resolving:
+            response["dnsResolving"] = dnsResolving(
```

Comment:
also results from this plz

Reply:
![image](https://github.com/intelowlproject/IntelOwl/assets/102874321/fdaca440-ee43-4529-8de6-da8405de93c2)
give nothing here...

Reply:
that's a problem cause I would have expected a populated answer. For instance, `xx.com` does exist.

The output should be like this: https://github.com/typosquatter/ail-typo-squatting?tab=readme-ov-file#dns-output Can you check the library closely pls?

Reply:
Figuring out the problem took quite some time. seems to work fine now. It just takes awfully lot of time.
attching a part of the output

```

{'k-com.xn--ogbpf8fl': {'NotExist': True}, 'k-com.xn--otu796d': {'NotExist': True}, 'k-com.xn--p1acf': {'NotExist': True}, 'k-com.xn--p1ai': {'NotExist': True}, 'k-com.xn--pgbs0dh': {'NotExist': True}, 'k-com.xn--pssy2u': {'NotExist': True}, 'k-com.xn--q7ce6a': {'NotExist': True}, 'k-com.xn--q9jyb4c': {'NotExist': True}, 'k-com.xn--qcka1pmc': {'NotExist': True}, 'k-com.xn--qxa6a': {'NotExist': True}, 'k-com.xn--qxam': {'NotExist': True}, 'k-com.xn--rhqv96g': {'NotExist': True}, 'k-com.xn--rovu88b': {'NotExist': True}, 'k-com.xn--rvc1e0am3e': {'NotExist': True}, 'k-com.xn--s9brj9c': {'NotExist': True}, 'k-com.xn--ses554g': {'NotExist': True}, 'k-com.xn--t60b56a': {'NotExist': True}, 'k-com.xn--tckwe': {'NotExist': True}, 'k-com.xn--tiq49xqyj': {'NotExist': True}, 'k-com.xn--unup4y': {'NotExist': True}, 'k-com.xn--vermgensberater-ctb': {'NotExist': True}, 'k-com.xn--vermgensberatung-pwb': {'NotExist': True}, 'k-com.xn--vhquv': {'NotExist': True}, 'k-com.xn--vuq861b': {'NotExist': True}, 'k-com.xn--w4r85el8fhu5dnra': {'NotExist': True}, 'k-com.xn--w4rs40l': {'NotExist': True}, 'k-com.xn--wgbh1c': {'NotExist': True}, 'k-com.xn--wgbl6a': {'NotExist': True}, 'k-com.xn--xhq521b': {'NotExist': True}, 'k-com.xn--xkc2al3hye2a': {'NotExist': True}, 'k-com.xn--xkc2dl3a5ee0h': {'NotExist': True}, 'k-com.xn--y9a3aq': {'NotExist': True}, 'k-com.xn--yfro4i67o': {'NotExist': True}, 'k-com.xn--ygbi2ammx': {'NotExist': True}, 'k-com.xn--zfr164b': {'NotExist': True}, 'k-com.xxx': {'NotExist': True}, 'k-com.xyz': {'NotExist': True}, 'k-com.yachts': {'NotExist': True}, 'k-com.yahoo': {'NotExist': True}, 'k-com.yamaxun': {'NotExist': True}, 'k-com.yandex': {'NotExist': True}, 'k-com.ye': {'NotExist': True}, 'k-com.yodobashi': {'NotExist': True}, 'k-com.yoga': {'NotExist': True}, 'k-com.yokohama': {'NotExist': True}, 'k-com.you': {'NotExist': True}, 'k-com.youtube': {'NotExist': True}, 'k-com.yt': {'NotExist': True}, 'k-com.yun': {'NotExist': True}, 'k-com.za': {'NotExist': True}, 'k-com.zappos': {'NotExist': True}, 'k-com.zara': {'NotExist': True}, 'k-com.zero': {'NotExist': True}, 'k-com.zip': {'NotExist': True}, 'k-com.zm': {'NotExist': True}, 'k-com.zone': {'NotExist': True}, 'k-com.zuerich': {'NotExist': True}, 'k-com.zw': {'NotExist': True}, 'k-com': {'NotExist': True}}
```

Reply:
cool man! :)